### ISSUE TITLE
[BUG] Failed to connect to headset display (496)

### ISSUE THREAD

**Issue by Martmists-GH** [ValveSoftware/SteamVR-for-Linux#636]
**Describe the bug**
It seems to fail to connect no matter what I try.

**To Reproduce**
1. Connect headset
2. Click "Enter VR Mode"

**Expected behavior**
VR works as expected

**System Information (please complete the following information):**
- Distribution: Arch
- SteamVR version: 2.0.10
- Steam client version: 1698777785
- Opted into Steam client beta?: No(?)
- Graphics driver version:  545.29.02
- Gist for SteamVR System Information: No "SteamVR beta" option? Found System Report instead:
[SteamVR-2023-11-09-PM_06_20_06.txt](https://github.com/ValveSoftware/SteamVR-for-Linux/files/13310993/SteamVR-2023-11-09-PM_06_20_06.txt)

**Screenshots**
N/A

**Additional context**
I'm also experiencing issues with xrdesktop crashing with the following:
```
dbus property change: true
vrcompositor-launcher.sh[175844]: === Thu Nov  9 06:22:52 PM CET 2023 ===
vrcompositor-launcher.sh[175844]: Relaunch under scout LDLP runtime.
vrcompositor-launcher.sh[175844]: exec /home/mart/.steam/root/ubuntu12_32/steam-runtime/run.sh /home/mart/.local/share/Steam/steamapps/common/SteamVR/bin/linux64/vrcompositor-launcher.sh
vrcompositor-launcher.sh[175844]: Detected scout LDLP runtime.
vrcompositor-launcher.sh[175844]: exec /home/mart/.local/share/Steam/steamapps/common/SteamVR/bin/linux64/vrcompositor-launcher
Using vrcompositor capability proxy
Failed to raise ambient cap
Launching /home/mart/.local/share/Steam/steamapps/common/SteamVR/bin/linux64/vrcompositor
Using breakpad crash handler
Setting breakpad minidump AppID = 250820
Forcing breakpad minidump interfaces to load
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
11/09 18:22:52 Init: Installing breakpad exception handler for appid(250820)/version(2.0.10)/tid(175844)

** (process:175800): WARNING **: 18:22:52.946: [XR_ERROR_RUNTIME_FAILURE] Failed to create session
Could not init VR session.

** (process:175800): WARNING **: 18:22:52.953: Failed to create gxr context, unable to create g3k context
Could not init VR runtime.
Failed to initialize xrdesktop!
Usually this is caused by a problem with the VR runtime.
```
though I'm not sure if this is related or not.

**Comment by Darth-Ender** [comment]
I'm getting this error in debian unstable as well.   Beta steamvr, normal steamvr, beta client, normal client. In wayland and in Xorg. Nothing i try seems to want to let things work like they used to with my Vive. 

I dont have xrdesktop errors and such,  just the 496 can't connect to headset / headset not found error from what it looks like. Nothing else seems to stand out error-wise.  

kernel is 6.6
Mesa 23.2.1 

Not sure what other info is relevant. everything worked perfectly like a month ago.

**Comment by Darth-Ender** [comment]
FYI, i resolved the connecting to headset issue by replacing the 3-in-1 cable on the headset.  My issue was definitely the HDMI cable the headset uses.  Swapping a different hdmi cable brought it right back.

You can verify if this issue is your issue by using xrandr (even in wayland).  if the second display doesn't show 2160x1200 as a detected resolution (or it shows almost nothing at all or nothing at all) ..then it's likely the hdmi cable has degraded to the point of not being functional.

**Comment by diademiemi** [comment]
I'm still getting the issue and the xrandr output seems fine. It worked 2 days ago

**Comment by diademiemi** [comment]
After further investigation I think it's a hardware issue for me too, I just reached out to customer support

**Comment by diademiemi** [comment]
After more troubleshooting, adding https://github.com/ValveSoftware/steam-devices/blob/master/60-steam-vr.rules to /etc/udev/rules.d fixed the issue. Very strange how it worked for a day without them

**Comment by Darth-Ender** [comment]
I purchased a new 3 in one cable.  The display worked for a day.  Then back to not working.   edid decode data looks the same as any others i've been able to find online.    I tried manually adding the mode in xrandr that the vive uses but no dice there.   I've got no idea why this is not working after working just fine for so long.  Unless there is some other underlying hardware or amd firmware issue.   I even added the udev rules linked above and still zip. 

Debian unstable 
Xorg: 21.1.9
vulkan: 1.3.268
glx:  AMD Radeon RX 7900 XT (gfx1100, LLVM 16.0.6, DRM 3.56, 6.7.0-rc2
(tried in X and wayland 1.22.0-2.1 (kde plasma ))

I keep coming to this issue with xrandr not being able to see the expected mode.  (vive is on output 0x53)
vrcompositor.txt 
Thu Nov 23 2023 16:39:36.976210 [Info] - Looking for direct display through RandR
Thu Nov 23 2023 16:39:36.976220 [Info] -  - Root 0x6ce
Thu Nov 23 2023 16:39:36.981920 [Info] -    - Output 0x52 - 44 modes, 1 preferred
Thu Nov 23 2023 16:39:36.981940 [Info] -      - Mode 0 0x57 2560x1440 (looking for 2160x1200)
Thu Nov 23 2023 16:39:36.981959 [Info] -    - Output 0x53 - 6 modes, 0 preferred
Thu Nov 23 2023 16:39:36.981976 [Info] -    - Output 0x54 - 0 modes, 0 preferred
Thu Nov 23 2023 16:39:36.981993 [Info] -    - Output 0x55 - 0 modes, 0 preferred
Thu Nov 23 2023 16:39:36.982000 [Info] - Tried to find direct display through RandR: (nil)
Thu Nov 23 2023 16:39:36.982005 [Info] - Looking for direct display through Vulkan WSI
Thu Nov 23 2023 16:39:36.982011 [Info] - Tried to find direct display through Vulkan WSI: (nil)


Below is the dmesg output related to amdgpu.   

[    1.097925] [drm] amdgpu kernel modesetting enabled.
[    1.100352] amdgpu: Virtual CRAT table created for CPU
[    1.100358] amdgpu: Topology: Add CPU node
[    1.100463] amdgpu 0000:0c:00.0: enabling device (0006 -> 0007)
[    1.104857] amdgpu 0000:0c:00.0: amdgpu: Fetched VBIOS from VFCT
[    1.104859] amdgpu: ATOM BIOS: 113-D70201-810006
[    1.104983] amdgpu 0000:0c:00.0: amdgpu: CP RS64 enable
[    1.105111] amdgpu 0000:0c:00.0: [drm:0xffffffffc0309296] JPEG decode is enabled in VM mode
[    1.105237] amdgpu 0000:0c:00.0: amdgpu: Trusted Memory Zone (TMZ) feature not supported
[    1.105255] amdgpu 0000:0c:00.0: amdgpu: MEM ECC is not presented.
[    1.105255] amdgpu 0000:0c:00.0: amdgpu: SRAM ECC is not presented.
[    1.105265] amdgpu 0000:0c:00.0: amdgpu: VRAM: 20464M 0x0000008000000000 - 0x00000084FEFFFFFF (20464M used)
[    1.105267] amdgpu 0000:0c:00.0: amdgpu: GART: 512M 0x00007FFF00000000 - 0x00007FFF1FFFFFFF
[    1.105329] [drm] amdgpu: 20464M of VRAM memory ready
[    1.105330] [drm] amdgpu: 32108M of GTT memory ready.
[    1.105720] amdgpu 0000:0c:00.0: amdgpu: Will use PSP to load VCN firmware
[    1.317926] amdgpu 0000:0c:00.0: amdgpu: RAP: optional rap ta ucode is not available
[    1.317929] amdgpu 0000:0c:00.0: amdgpu: SECUREDISPLAY: securedisplay ta ucode is not available
[    1.317961] amdgpu 0000:0c:00.0: amdgpu: smu driver if version = 0x0000003d, smu fw if version = 0x0000003f, smu fw program = 0, smu fw version = 0x004e6601 (78.102.1)
[    1.317967] amdgpu 0000:0c:00.0: amdgpu: SMU driver if version not matched
[    1.499276] amdgpu 0000:0c:00.0: amdgpu: SMU is initialized successfully!
[    3.327727] amdgpu 0000:0c:00.0: [drm:0xffffffffc030941b] JPEG decode initialized successfully.
[    3.328979] kfd kfd: amdgpu: Allocated 3969056 bytes on gart
[    3.328993] kfd kfd: amdgpu: Total number of KFD nodes to be created: 1
[    3.329015] amdgpu: Virtual CRAT table created for GPU
[    3.329289] amdgpu: Topology: Add dGPU node [0x744c:0x1002]
[    3.329292] kfd kfd: amdgpu: added device 1002:744c
[    3.329306] amdgpu 0000:0c:00.0: amdgpu: SE 6, SH per SE 2, CU per SH 8, active_cu_number 84
[    3.329312] amdgpu 0000:0c:00.0: amdgpu: ring gfx_0.0.0 uses VM inv eng 0 on hub 0
[    3.329315] amdgpu 0000:0c:00.0: amdgpu: ring comp_1.0.0 uses VM inv eng 1 on hub 0
[    3.329317] amdgpu 0000:0c:00.0: amdgpu: ring comp_1.1.0 uses VM inv eng 4 on hub 0
[    3.329319] amdgpu 0000:0c:00.0: amdgpu: ring comp_1.2.0 uses VM inv eng 6 on hub 0
[    3.329321] amdgpu 0000:0c:00.0: amdgpu: ring comp_1.3.0 uses VM inv eng 7 on hub 0
[    3.329323] amdgpu 0000:0c:00.0: amdgpu: ring comp_1.0.1 uses VM inv eng 8 on hub 0
[    3.329325] amdgpu 0000:0c:00.0: amdgpu: ring comp_1.1.1 uses VM inv eng 9 on hub 0
[    3.329327] amdgpu 0000:0c:00.0: amdgpu: ring comp_1.2.1 uses VM inv eng 10 on hub 0
[    3.329329] amdgpu 0000:0c:00.0: amdgpu: ring comp_1.3.1 uses VM inv eng 11 on hub 0
[    3.329331] amdgpu 0000:0c:00.0: amdgpu: ring sdma0 uses VM inv eng 12 on hub 0
[    3.329333] amdgpu 0000:0c:00.0: amdgpu: ring sdma1 uses VM inv eng 13 on hub 0
[    3.329335] amdgpu 0000:0c:00.0: amdgpu: ring vcn_unified_0 uses VM inv eng 0 on hub 8
[    3.329337] amdgpu 0000:0c:00.0: amdgpu: ring vcn_unified_1 uses VM inv eng 1 on hub 8
[    3.329339] amdgpu 0000:0c:00.0: amdgpu: ring jpeg_dec uses VM inv eng 4 on hub 8
[    3.329341] amdgpu 0000:0c:00.0: amdgpu: ring mes_kiq_3.1.0 uses VM inv eng 14 on hub 0
[    3.332435] amdgpu 0000:0c:00.0: amdgpu: Using BACO for runtime pm
[    3.332642] [drm] Initialized amdgpu 3.56.0 20150101 for 0000:0c:00.0 on minor 0

**Comment by caseif** [comment]
If it is a hardware issue in my case, it seems there's still a software component at play. I'm able to get the headset working on a separate Windows machine, but twice it gave a spurious message about the cable becoming disconnected at the headset, pressing the connection firmly resolved it though. Perhaps there's an issue with the hardware that SteamVR on Windows is somehow able to compensate for?

**Comment by Darth-Ender** [comment]
i think I've resolved my issue by bypassing the link box for the hdmi connection.   it might even be that the original cable was fine and it was always just the link box going bad.  xrandr immediately showed all the correct resolutions and steamvr worked perfectly once bypassed.

**Comment by caseif** [comment]
I was encountering this issue intermittently in a Wayland session, then around a month ago it started occurring consistently in Wayland, then some time later it started occurring consistently in X11 as well. I found that renaming `steamvr.vrsettings` opening SteamVR to generate a new one, closing SteamVR, and then restoring the original version solved the problem for me, at least for the time being.

**Comment by Adamcake** [comment]
Valve doesn't read their issue trackers, so posting here is akin to screaming into the void. Hopefully one day they'll hire their first ever competent developer. Until then at least, if someone finds a workaround for this, let me know.

**Comment by Mon-Ouie** [comment]
I'm a bit confused what happened, I experienced this issue for the first time ever today using my Index. The only thing I did since SteamVR last worked was I tried ALVR with a Quest 3 (I did make sure `~/.config/openvr/openvrpaths.vrpath` was back to normal when switching back to using the Index).

*Things I tried that did not work*:

1. Deleting `steamvr.vrsettings`
2. Adding the udev rules
3. Unplugging the Index (also I was able to run VR on Windows so I don't think the cable is faulty).
4. Rebooting after doing each step

**What did work eventually**: switching from the normal branch to the 2.2.2 beta. I would then experience issue #638, but everything works after switching back to the main branch. I'm not sure if ALVR messed up some files in my installation or if it's a coincidence. Maybe try a clean reinstallation of SteamVR if you encounter this issue?

**Comment by MegaShinySnivy** [comment]
> having this issue on a valve index, arch Linux, kde plasma wayland, latest kernel, steam client beta, steamvr (beta, and stable). I pray that the steamvr team leaves the "temp_v1.27.5" beta branch available until this is resolved...

I'm having the same issue. It only works under temp_v1.27.5, which tells me this isn't a hardware issue, its a software issue. Exact same setup too. The only way I could get it to work was that beta under x11, I had to ditch wayland. Also, any proton VR games (that is to say, all of them) crash steam on launch.

**Comment by Darth-Ender** [comment]
i think it's safe to say this error can be caused by a few different
reasons.  my issue was 100% hardware.  bypassing the extension box for hdmi
resolved my issue and i didn't need to ditch wayland or the latest version
of everything that I'm using with my 7900xt.

**Comment by MegaShinySnivy** [comment]
> i think it's safe to say this error can be caused by a few different reasons. my issue was 100% hardware. bypassing the extension box for hdmi resolved my issue and i didn't need to ditch wayland or the latest version of everything that I'm using with my 7900xt.

Are you recommending we open a new issue then?

**Comment by ion-ize** [comment]
I am having the exact same issue and none of the fixes listed for othere here are working for me.

Things I have tried:

1. Deleting steamvr.vrsettings
2. Adding the udev rules
3. Unplugging the Index (also I was able to run VR on Windows just 5 days ago so I don't think the cable is faulty)
4. Switching from the normal branch to the 2.6.2 beta
5. Switching the Steam client to the stable branch from the families beta branch
6. Rebooting after doing each step

System Information (please complete the following information):

Distribution: [Pop!_OS 22.04 LTS]
SteamVR version: [SteamVR beta 2.6.2 (also tried the stable branch)]
Steam client version: [1718395788 Steam Families Beta]
Opted into Steam client beta?: [Yes (tried both with and without the beta)]
Graphics driver version: [Nvidia 550.63]
GPU ~ RTX 2080 TI, CPU ~ AMD® Ryzen 9 3900x 12-core processor × 24, MB ~ ROG Crosshair VIII Hero
Gist for SteamVR System Information: https://gist.github.com/Crimsonize/a2eeab857b6af847b1478384b02f8778

**Comment by Darth-Ender** [comment]
the only way to be sure it's not a hardware problem is to swap in
replacements until it starts working and isolate the part you need to
replace or discover that it doesn't work if everything is replaced (which
may indicate software issue)

my situation was hardware related.   i bypass the extender box and plug the
hdmi directly into my video card.   also got a new triple cable.

**Comment by JerryLuce** [comment]
Just writing to say I'm also having this issue. New install of Manjaro. Wayland, KDE-Plasma 6.
Works perfectly fine on my old windows install.  Tried using the betas and the release of SteamVR. Tried all the above troubleshooting to no avail.

**Comment by failboat78** [comment]
Same problem persists with steamvr beta 2.9.5.

But when I run normal version (2.8.8) it works!

**Comment by benbot** [comment]
Nixos, KDE plasma 6 wayland, rtx 4090 Valve Index.

Works on 2.8.8, but not the latest

**Comment by Martmists-GH** [comment]
I managed to sort of make it work, but I have to unplug the headset before booting, then plug it in before launching steam. I think there's a chance the nvidia driver might be taking control of the device and SteamVR can't get direct control?

**Comment by Renari** [comment]
> I managed to sort of make it work, but I have to unplug the headset before booting, then plug it in before launching steam. I think there's a chance the nvidia driver might be taking control of the device and SteamVR can't get direct control?

This was my issue as well got it working by doing the same thing, this also makes the odd 640x480 display not show up in plasma.

**Comment by wodxtex** [comment]
@Renari  sounds like you have the same issue... i think i fixed it by enabling  HDCP 1.4 Legacy Compatibility and DisplayPort Training mode to Compatibility mode A seems to fix it and doesn't display the 640x480 display in plasma no more.

**Comment by PureTryOut** [comment]
Same problem on an AMD RX 9070XT, so not NVIDIA. What solved it for me was having the DisplayPort of the headset (Valve Index) unplugged during boot and only connecting it once booted into a logged-in (KDE Plasma on Wayland) desktop.
Now the only remaining issue is the speakers of the headset not working at all but at least this is a workable setup now when using a separate headset 🎉

**Comment by Martmists-GH** [comment]
Another change that seemed to make an improvement (not 100% tested) is enabling kernel mode setting on the nvidia/amdgpu drivers.

**Comment by vrbandwagon** [comment]
> Another change that seemed to make an improvement (not 100% tested) is enabling kernel mode setting on the nvidia/amdgpu drivers.

How does one do that exactly? I would like to try that, as the unplugging during boot doesn't seem to work for me. Also, is the mesa driver downgrade still relevant, or has this been addressed with the latest drivers?

**Comment by Martmists-GH** [comment]
> > Another change that seemed to make an improvement (not 100% tested) is enabling kernel mode setting on the nvidia/amdgpu drivers.
> 
> How does one do that exactly? I would like to try that, as the unplugging during boot doesn't seem to work for me. Also, is the mesa driver downgrade still relevant, or has this been addressed with the latest drivers?

I'd love to do more testing, but the recent mesa driver update causes my testing to be blocked by https://github.com/ValveSoftware/SteamVR-for-Linux/issues/836
